gravitee-am-service/src/main/java/io/gravitee/am/service/impl/I18nDictionaryService.java

```diff
@@ -0,0 +1,254 @@
+                        return dictionary;
+                    }
+                })
+                .flatMap(repository::create)
+                .flatMap(dictionary -> {
+                    Event event = new Event(Type.I18N_DICTIONARY, new Payload(dictionary.getId(), dictionary.getReferenceType(), dictionary.getReferenceId(), Action.CREATE));
+                    return eventService
+                            .create(event)
+                            .flatMap(createdEvent -> Single.just(dictionary));
+                })
+                .onErrorResumeNext(ex -> {
+                    if (ex instanceof AbstractManagementException) {
+                        return Single.error(ex);
+                    } else {
+                        String msg = "An error occurred while trying to create an i18n dictionary";
+                        LOGGER.error(msg, ex);
+                        return Single.error(new TechnicalManagementException(msg, ex));
+                    }
+                })
+                .doOnSuccess(dictionary -> auditService.report(AuditBuilder
+                                                                       .builder(DictionaryAuditBuilder.class)
+                                                                       .principal(principal)
+                                                                       .type(EventType.I18N_DICTIONARY_CREATED)
+                                                                       .dictionary(dictionary)))
+                .doOnError(throwable -> auditService.report(AuditBuilder
+                                                                    .builder(DictionaryAuditBuilder.class)
+                                                                    .principal(principal)
+                                                                    .type(EventType.I18N_DICTIONARY_CREATED)
+                                                                    .throwable(throwable)));
+    }
+
+    public Maybe<I18nDictionary> findByName(ReferenceType referenceType, String referenceId, String name) {
+        LOGGER.debug("Find dictionary by {} and name: {} {}", referenceType, referenceId, name);
+        return repository.findByName(referenceType, referenceId, name)
+                         .onErrorResumeNext(ex -> {
+                             String msg = "An error occurred while trying to find a dictionary using its name:";
+                             LOGGER.error(msg + " {} for the {} {}", name, referenceType, referenceId, ex);
+                             return Maybe.error(new TechnicalManagementException(
+                                     String.format(msg + " %s for the %s %s", name, referenceType, referenceId), ex));
+                         });
+    }
+
+    public Single<I18nDictionary> update(ReferenceType referenceType, String referenceId, String id, UpdateDictionary updateDictionary) {
```

Comment:
As we create an audit entry for the create action, we should also do it for the Update